server/controllers/Project.js

```diff
@@ -0,0 +1,72 @@
+const { User, Project } = require('../models');
+const upload = require('../services/file-upload');
+const singleUpload = upload.single('image');
+const multiUpload = upload.array('images');
+
+
+exports.imageUpload = (req, res) => {
+    singleUpload(req, res, function (err) {
+        if (err) return res.status(422).json({ errors: [{ title: 'File Upload Error', detail: err.message }] });
+        return res.json({ 'imageUrl': req.file.location })
+    })
+}
+
+exports.projectById = (req, res, next, id) => {
+    Project.findById(id)
+        .populate('user')
+        .exec((err, project) => {
+            if (err || !project) {
+                return res.status(400).json({
+                    error: 'Project not found.'
+                });
```

Comment:
Nothing blocking this from being merged, but we should discuss how we send error json, IE as array or single object.

Reply:
👍 in the past I've created helper functions to send error messaging